### ISSUE TITLE
Design Intent - Accessibility Feedback - AE Profile | Notification Settings | Appointment Reminders PHI/PII

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#92376]
## Next Steps for the VFS team

- [ ] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [ ] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

- Thank you for your patience on this!
- @jeana-adhoc, please feel free to jump in with thoughts!

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [ ] **Should:** Consider structuring this as a single conditional reveal, with one question on receiving notifications and one question on notifications containing PHI. Something like:
   ```
   Health appointment reminders
   [ ] Notify me by email
   [ ] Notify me by text
   
       Do you want health appointment reminders to include facility and clinic details?
       You can opt in to include facility and clinic details in your reminders. If you opt in, we'll include the name and location of your health facility and clinic. The clinic name may include the type of care you're receiving, like "cardiology."
       [ ] Include facility and clinic details in email notifications
       [ ] Include facility and clinic details in text notifications
   ```
   ... where the options displayed in the second question are determined by which boxes are checked in the first, ie. "Include facility and clinic details in email notifications" is only there if "Notify me by email" is checked. I'm definitely not a content writer, but something along these lines might work.
   
   This sidesteps the multiple nested fieldsets problem, and is basically consistent with the existing guidance on conditional reveals. It also breaks the interaction cleanly into two questions: Do you want to receive notifications? And should the notifications include possible PHI? This is definitely my preference if it works for you!
- [ ] **Must:** If the recommendation above doesn't work for you, then here are some requirements for implementing what you currently have:
   - [ ] The conditionally-revealed inputs need to have a clear question or prompt associated with them, rather than just the label on the checkbox itself, to provide clear context that this is a new follow-up question and not another possible option for the initial prompt. 
   - [ ] The follow-up question needs to be phrased so that it's grammatically independent of the initial prompt --- if you just said it out loud to someone, it would make sense --- and distinct for each notification option. (Meaning, the PHI question for text and the PHI question for email need to be clearly for either the text or email options, not the same question repeated twice.)
   - [ ] The follow-up question can be visually hidden if needed, but it still needs to be programmatically associated with the input. I've thought about different ways to do this and I think a nested `fieldset`/`legend` pair makes the most sense, and if you visually hide it just put the `sr-only` class on the `legend`.
   
   And if you go that route, I'll recommend one small change change to the design system guidance.
   
   Current version:
   > Conditionally revealed fields can be used if the following conditions are met:
   > 
   > 1. There should only be one reveal on a page.
   > 2. When the revealed trigger is selected, you must be able to tab directly into the newly revealed field (Which is why we’ve put the “other” question last.)
   > 3. The newly revealed question field must be understood by itself. For example, don’t just say “Other”. Instead, say:
   > > Since your relationship with the veteran was not listed, please describe it here
   
   New version:
   > Conditionally revealed fields can be used if the following conditions are met:
   > 
   > 1. There should only be one reveal on a page. **Exceptions may be granted for repeated instances of the same revealed input, where appropriate.**
   > 2. When the revealed trigger is selected, you must be able to tab directly into the newly revealed field (Which is why we’ve put the “other” question last.)
   > 3. The newly revealed question field must be understood by itself. For example, don’t just say “Other”. Instead, say:
   > > Since your relationship with the veteran was not listed, please describe it here
   
   I think that should carve out a sufficient loophole for you here --- but I'd love it if we don't have to do that!

## Governance team actions
- [x] Format feedback as individual tasks (check boxes)
- [ ] Assign this ticket to the VFS team member that opened the Slack request
- [ ] Add the VFS team product label
- [ ] Add the VFS team the feature label (if applicable)
- [x] Add the touchpoint labels
- [x] Add the practice area labels
- [ ] Add the Collaboration Cycle initiative milestone

**Comment by twoangstroms** [comment]
@mtcA6 Can we serve different content in the conditionally revealed fields based on what the user selects in the static fields? What if a user selects "email", a field is revealed to ask about PHI/PII in email communications, and then the user also selects "text"?

**Comment by mtcA6** [comment]
@asiisii can we take a few min to check this tomorrow together? I'll send an invite

**Comment by twoangstroms** [comment]
I'll brainstorm how to word things in case that won't work.

**Comment by asiisii** [comment]
@mtcA6 yeah

**Comment by twoangstroms** [comment]
Sadly, I think there's a conflict state in Brian's otherwise good idea.

[x] Notify me by email
[ ] Notify me by text
     
     [ ] Include PHI/PII in email
     [x] Include PHI/PII in text

**Comment by twoangstroms** [comment]
@mtcA6 Ah, I read that as if the user clicked on one of the primary checkboxes (email or text), that determined what was conditionally revealed. You're saying that if the user clicked on "email" and that revealed the hint text and "include" checkbox, and _then_ the user clicked "text", more content would populate the revealed area? I'm not sure I've seen that behavior before but I haven't looked specifically for it.

**Comment by mtcA6** [comment]
@twoangstroms it doesn't really match the pattern from the UK example https://design-system.service.gov.uk/components/checkboxes/conditional-reveal/  

![image](https://github.com/user-attachments/assets/1102356c-e2b7-453c-bb9f-84d8affd029c)

and I also don't think it matches the design system, but that's why you, me, and Asiisii need to talk about it. [But mocking it up costs us next to nothing](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1724854864669/fd19195a421188d983c7395240861de29b87ed6d?wid=0-1725973413991) 🤷‍♂️ 

Brian also says:  **If the recommendation above doesn't work for you, then here are some requirements for implementing what you currently have:** where we could do what we mocked up for design intent but would need to take extra steps in our aria labels to make sure this is apt for screenreader users

**Comment by twoangstroms** [comment]
@mtcA6 Which "it" are you referring to above?

Yes, I was sketching all the iterations. That's how I came upon the situation I put in [above](https://github.com/department-of-veterans-affairs/va.gov-team/issues/92376#issuecomment-2339344339). And as I mentioned, I am working on text-centric options, as Brian suggested.

**Comment by twoangstroms** [comment]
@mtcA6 Maybe it's my morning brain, but was the "it" that doesn't match the UK pattern referring to what I put in a the comment I linked, or to what Brian proposed?

**Comment by ACParker89** [comment]
@twoangstroms In the interest of time, lets please mock both of these suggestions from Brian up so that we can make a decision on how to proceed. We should have them mocked this week, so we can make a decision and have a FE ticket ready for this work by Sprint 46 (beginning 9/25). Happy to noodle through them together on a call!

**Comment by mtcA6** [comment]
### Links from our call 
- https://design.va.gov/storybook/?path=/docs/uswds-va-checkbox-group--docs
- https://www.figma.com/community/file/1256660046319231869


### text for dan
- https://forum.figma.com/t/show-something-when-at-least-1-checkbox-is-checked/54074
to tie a conditional to a checkbox state take the following steps:
Create variant set with different check box states
Set up change to interactions for those checkbox variants
Bind those variant instances to a variable that matches the variant name
When the variant changes state, the variable bound to it will also update its value to match the variant change during the interaction. You can then set a conditional to trigger an action when the variable hits a certain value, in this case, the value that the variant has changed to in the interaction.

**Comment by ACParker89** [comment]
@twoangstroms can we discuss this today? do we need this?

**Comment by mtcA6** [comment]
The new team is opting to go the MUST route
